cache-config/testing/docker/docker-compose-ats-build.yml

```diff
@@ -27,11 +27,16 @@ services:
 
   trafficserver_build:
     environment:
-      - ATS_VERSION=8.1.x
```

Comment:
Why no longer define the ATS_VERSION environment variable in the compose file or an env file? I see that `run.sh` sets a default value for `ATS_VERSION` if `ATS_VERSION` is undefined, but if `ATS_VERSION` is meant to be passed to `run.sh`, a user should be able to see a sane default for `ATS_VERSION` without running `run.sh`, by running something like

```shell
docker-compose -f docker-compose-ats-build.yml run --rm trafficserver_build sh -c 'echo $ATS_VERSION'
```

Reply:
I moved it to the environment section of the GHA workflow/cache-config-tests.yml.  when we want to start testing against 9, we only need change the cache-config-tests.yml

Reply:
> I moved it to the environment section of the GHA workflow/cache-config-tests.yml thinking that it was made available to all runners.

Sure, it will be made available to all GitHub Actions runners, the T3C integration tests should continue to be usable outside of GitHub Action.

I see that there is a default set in `run.sh` itself:
https://github.com/apache/trafficcontrol/blob/a3f3e2045b86e734f52c043c59d9dfd73641b29e/cache-config/testing/docker/trafficserver/run.sh#L28
but now that means that users of the T3C integration tests would need to poke around in the scripts themselves to see what environment variables it uses. The compose file or an env file should exist to serve as a reference for users wanting to know what environment variables they can set, especially since the user can already learn about `CJOSE_URL`, `CJOSE_TAG`, `JANSSON_URL`, `JANSSON_TAG`, `OPENSSL_URL`, and `OPENSSL_TAG` from the compose file.

I do see the value in being able to set that environment variable from GitHub Actions. In order to do that:

1. The hard-coded `8.1.x` provided as the `branch` input for the *fetch-github-branch-sha* action should instead reference the `ATS_VERSION` environment variable:https://github.com/apache/trafficcontrol/blob/a3f3e2045b86e734f52c043c59d9dfd73641b29e/.github/workflows/cache-config-tests.yml#L103-L109
2. The `ATS_VERSION` environment variable should be referenced used for the *fetch-github-branch-sha* action's `branch` input in the *CDN-in-a-Box CI* workflow, as well:
    https://github.com/apache/trafficcontrol/blob/a3f3e2045b86e734f52c043c59d9dfd73641b29e/.github/workflows/ciab.yaml#L224-L230
3. The `ATS_VERSION` environment variable needs to be defined in for the *CDN in a Box CI* workflow:
    https://github.com/apache/trafficcontrol/blob/a3f3e2045b86e734f52c043c59d9dfd73641b29e/.github/workflows/ciab.yaml#L20-L21
4. (Already covered:) Removing a definition of `ATS_VERSION` from `docker-compose-ats-build.yml ` breaks `bin/ats-version.sh`, used by the CDN in a Box `ATS_DIST_RPM` makefile target:
    https://github.com/apache/trafficcontrol/blob/a3f3e2045b86e734f52c043c59d9dfd73641b29e/infrastructure/cdn-in-a-box/bin/ats-version.sh#L30
    So that you don't need to touch the `ats-version.sh`, I have opened #5945 so that CDN in a Box does not break when #5938 is merged.

Reply:
So a lot of this is me trying and testing different things in the GHA workflow.  So don't get too excited with me pushing up things to test in the workflow.  I'll let you know when I'm done ok.

Reply:
Sounds good, I'll wait to hear from you.